File: src/main/java/io/github/lncvrt/lncvrtbox/events/PrepareItemEnchantListener.java
package io.github.lncvrt.lncvrtbox.events;

import org.bukkit.Material;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.enchantment.PrepareItemEnchantEvent;
import org.bukkit.inventory.ItemStack;

public class PrepareItemEnchantListener implements Listener {
    @EventHandler
    public void onPrepareItemEnchant(PrepareItemEnchantEvent event) {
        ItemStack item = event.getItem();
        if (item.getType() == Material.DIAMOND_PICKAXE || item.getType() == Material.NETHERITE_PICKAXE) {
            event.setCancelled(true);
        }
    }
}

File: src/main/kotlin/xyz/lncvrt/galaxyboxpvp/events/PrepareItemEnchantListener.kt
package xyz.lncvrt.galaxyboxpvp.events

import org.bukkit.Material
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.enchantment.PrepareItemEnchantEvent

class PrepareItemEnchantListener : Listener {
    @EventHandler
    fun onPrepareItemEnchant(event: PrepareItemEnchantEvent) {
        val item = event.item
        if (item.type == Material.DIAMOND_PICKAXE || item.type == Material.NETHERITE_PICKAXE) event.isCancelled = true
    }
}